// issue-manager/issue.js

class Issue {
    constructor(id, title, status, assignee, created, updated, details) {
        this.id = id;
        this.title = title;
        this.status = status;
        this.assignee = assignee;
        this.created = created;
        this.updated = updated;
        this.details = details;
    };
};

var issue_1 = new Issue('001', 'Debug', 'Resolved', 'Bob Stuart', '02/02/2020', '02/03/2020', 'Syntax');
var issue_2 = new Issue('002', 'Revise', 'Closed', 'John Smith', '12/12/2020', '12/23/2020', 'Testing');
var issue_3 = new Issue('003', 'Test', 'New', 'Adam Clark', '09/03/2020', '09/15/2020', 'Error');
var issue_4 = new Issue('004', 'Review', 'Resolved', 'Edgar Lofton', '07/02/2020', '07/22/2020', 'New version');
var issue_5 = new Issue('005', 'Model_101', 'Closed', 'Athony Pilla', '01/02/2020', '01/03/2020', 'Implementation');
var issuesArray = [issue_1, issue_2, issue_3, issue_4, issue_5];


function issues() {
    return JSON.stringify(issuesArray) 
};

function create_issues(id, title, status, assignee, created, updated, details) {
    var newIssue =  new Issue(id, title, status, assignee, created, updated, details);
    issuesArray.push(newIssue);
};

function update_issue(issue_id, body) {
    var found = false;
    for(var i = 0; i < issuesArray.length; i++) {
        if(issue_id === issuesArray[i].id) {
            issuesArray[i] = body;
            found = true;
            break;
        }
    }
    if(found == false) {
        console.log(`Issue ${issue_id} not found in issues Array.`);
    }
}

function delete_issue(issue_id) {
    var found = false;
    for(var i = 0; i < issuesArray.length; i++) {
        if(issue_id === issuesArray[i].id) {
            var index = issuesArray.indexOf(issuesArray[i]);
            issuesArray.splice(index, 1);
            found = true;
            break;
        }
    }
    if(found == true) {
        console.log(`Issue ${issue_id} has been deleted from issues Array.`);
    }
    else {
        console.log(`Issue ${issue_id} not found.`);
    }
}


exports.issues = issues;
exports.create_issues = create_issues;
exports.update_issue = update_issue;
exports.delete_issue = delete_issue;
